Frontend/ui-customer/src/Flow.purs

```diff
@@ -3388,20 +3385,33 @@ metroTicketDetailsFlow = do
   case flow of 
     BACK_TO_SEARCH_METRO_LOCATION -> metroTicketBookingFlow
     GO_BACK_TO_HOME_SCREEN -> homeScreenFlow
+    GO_TO_MY_METRO_TICKETS_FLOW ->  metroMyTicketsFlow
     _ -> metroTicketDetailsFlow
 
 metroMyTicketsFlow :: FlowBT String Unit
 metroMyTicketsFlow = do
   logField_ <- lift $ lift $ getLogFields
   flow <- UI.metroMyTicketsScreen 
   case flow of 
-    GO_TO_METRO_TICKET_DETAILS_FLOW bookingStatusResp -> do
-      modifyScreenState $ MetroTicketDetailsScreenStateType (\metroTicketDetailsState -> metroTicketDetailsTransformer bookingStatusResp metroTicketDetailsState)
-      modifyScreenState $ MetroTicketDetailsScreenStateType (\slsState -> slsState{props{previousScreenStage = ST.MetroTicketStatusStage}})
-      metroTicketDetailsFlow
+    GO_TO_METRO_TICKET_DETAILS_FLOW  bookingId -> do
+      (GetMetroBookingStatusResp resp) <- Remote.getMetroStatusBT bookingId
+      let (MetroTicketBookingStatus metroTicketBookingStatus) = resp
+      if(metroTicketBookingStatus.status == "CONFIRMED") then do
+        modifyScreenState $ MetroTicketDetailsScreenStateType (\metroTicketDetailsState -> metroTicketDetailsTransformer resp metroTicketDetailsState)
+        modifyScreenState $ MetroTicketDetailsScreenStateType (\slsState -> slsState{props{previousScreenStage = ST.MetroMyTicketsStage}})
+        metroTicketDetailsFlow
+      else if (metroTicketBookingStatus.status == "PAYMENT_PENDING" || metroTicketBookingStatus.status == "FAILED") then do
+        modifyScreenState $ MetroTicketStatusScreenStateType (\metroTicketStatusScreen -> metroTicketStatusTransformer resp "" metroTicketStatusScreen)
```

Comment:
No need to pass empty string to `metroTicketStatusTransformer`, If it is not being used.